tests/test_proxy_connect.py

```diff
@@ -67,6 +81,7 @@ def test_https_connect_tunnel(self):
             yield crawler.crawl(self.mockserver.url("/status?n=200", is_secure=True))
         self._assert_got_response_code(200, l)
 
+    @pytest.mark.xfail(reason='mitmproxy gives an error for noconnect requests')
```

Comment:
What error is that? Is it some feature missing in mitmproxy, which was present before? If so, it'd be good to open a ticket in their issue tracker, and add a link  here.

Reply:
"Invalid request scheme: https" like in https://github.com/mitmproxy/mitmproxy/issues/848.

See https://github.com/scrapy/scrapy/issues/2545#issuecomment-532153176

Note that the mitmproxy issue above is open but I'm not 100% sure it's the same use case. Actually, I'm not sure *what* is the use case for the Scrapy code to skip CONNECT.

Reply:
https://github.com/scrapy-plugins/scrapy-crawlera/pull/44

Reply:
@kmike I've added a comment about this as requested